lang/rust/avro/src/writer.rs

```diff
@@ -109,15 +111,27 @@ impl<'a, W: Write> Writer<'a, W> {
     pub fn append_value_ref(&mut self, value: &Value) -> AvroResult<usize> {
         let n = self.maybe_write_header()?;
 
-        write_value_ref(self.schema, value, &mut self.buffer)?;
+        // Lazy init for users using the builder pattern with error throwing
+        match self.resolved_schema {
+            Some(ref rs) => {
+                write_value_ref_resolved(rs, value, &mut self.buffer)?;
+                self.num_values += 1;
 
-        self.num_values += 1;
+                if self.buffer.len() >= self.block_size {
+                    return self.flush().map(|b| b + n);
+                }
 
-        if self.buffer.len() >= self.block_size {
-            return self.flush().map(|b| b + n);
+                Ok(n)
+            }
+            None => {
+                if !self.attempted_resolve {
```

Comment:
Is it possible the value to be already `true` ?
Maybe we can remove the `if` ?!

Reply:
I've removed the check.

Reply:
Oh right, this check does nothing based on the design I ended with 😅 . I was trying to differentiate between builder built writers and constructor built writers. Thank you!

Reply:
I believe we should be able to remove the bool all together